src/aosm/azext_aosm/cli_handlers/onboarding_base_handler.py

```diff
@@ -89,6 +92,7 @@ def generate_config(self, output_file: str | None = None):
 
     def build(self):
         """Build the definition."""
+        self.pre_validate_build()

```

Comment:
My thinking was: this is not implemented in any of the handlers and doesn't have to be (because if you don't implement it, it doesn't do anything). But I could see a future where other handlers would want to do some validation before building, which is why I think it is appropriate in base